util/testutil/dockerd/client/options.go

```diff
@@ -0,0 +1,29 @@
+package client
+
+import (
+	"net/http"
+
+	"github.com/docker/go-connections/sockets"
+	"github.com/pkg/errors"
+)
+
+// Opt is a configuration option to initialize a [Client].
+type Opt func(*Client) error
+
+// WithHost overrides the client host with the specified one.
+func WithHost(host string) Opt {
+	return func(c *Client) error {
+		hostURL, err := ParseHostURL(host)
+		if err != nil {
+			return err
+		}
+		c.host = host
+		c.proto = hostURL.Scheme
+		c.addr = hostURL.Host
+		c.basePath = hostURL.Path
+		if transport, ok := c.client.Transport.(*http.Transport); ok {
+			return sockets.ConfigureTransport(transport, c.proto, c.addr)
```

Comment:
Do we really need this? Ideally we should get rid of this `sockets` pkg as well. I don't see it doing anything for us. Atm it looks like we are calling it twice, once in the `NewClientWithOpts` constructor and then in here. While the actual `client.Transport` value is not controllable at all and is always hardcoded.